src/rgw/store/sfs/sfs_user.cc

```diff
@@ -30,7 +30,8 @@ int SFStore::get_user_by_access_key(const DoutPrefixProvider *dpp,
   int err = 0;
   rgw::sal::sfs::sqlite::SQLiteUsers sqlite_users(dpp->get_cct());
   auto db_user = sqlite_users.getUserByAccessKey(key);
-  if (db_user) {  
+  if (db_user) {
+    db_user->uinfo.user_id.tenant = db_user->uinfo.user_id.id;  // TODO Remove this when ACL is implemented
     db_user->uinfo.user_id.id = "";   // TODO Remove this when ACL is implemented
```

Comment:
I'm not a fan of TODO comments remembering that something must be changed in the future.
Btw, can you explain me a bit this logic on `tenant` and `id` ? why the assignment and the id blanking?

Reply:
Yeah, me neither, but I needed to add this hack and that's the way I found to warn about it.

The problem is basically that if `user_id.id` is set, rgw tries to read attributes for ACLs and nothing works.
The original `sfs` implementation (returning always the default user) had `user_id.id == ""`, that's why it also skipped that check. 

In my previous PR I was simply setting `user_id.id = ""` but in this case, after adding the rest api stuff, we need that id for removing the user, loading, etc...  So I'm storing that id in the `tenant` field for now...

I'll start trying to add the ACLs stuff or, at least, make it work without those ugly hacks in next PR (or if I find a solution in a couple of days or less will add it to this one), so that crap should not be there for long...

Reply:
ok, undestood, thanks. Yes, this is a bit hacky patch.
Can't we wait maybe having a more consistent implementation with ACLs?
My fear is that having TODOs around the code, in general, could lead to forget something dirty somewhere.

Reply:
Cool, I found out that just returning the attributes of the user from the DB makes rgw to be happy.
So, ugly hack removed.